hrms/hr/doctype/goal/goal.js

```diff
@@ -48,19 +48,26 @@ frappe.ui.form.on("Goal", {
 
 	add_custom_buttons(frm) {
 		if (!frm.doc.__islocal) {
+		  if (frm.doc.status !== "Closed") {
+			frm.add_custom_button(__("Close"), () => {
+			  frm.set_value("status", "Closed");
+			  frm.save();
+			});
+
 			if (frm.doc.status !== "Archived") {
-				frm.add_custom_button(__("Archive"), () => {
-					frm.set_value("status", "Archived");
-					frm.save();
-				});
+			  frm.add_custom_button(__("Archive"), () => {
+				frm.set_value("status", "Archived");
+				frm.save();
+			  });
 			} else {
-				frm.add_custom_button(__("Unarchive"), () => {
-					frm.set_value("status", "");
-					frm.save();
-				});
+			  frm.add_custom_button(__("Unarchive"), () => {
+				frm.set_value("status", "");
+				frm.save();
+			  });
 			}
+		  }
```

Comment:
Let's combine this under a Status dropdown button. 

<table>
<tr>
 <td>Current Status
 <td>Buttons
<tr>
 <td>Pending or In Progress
 <td>Archive and Close options
<tr>
 <td>Archived
 <td>Unarchived
<tr>
 <td>Closed
 <td>Reopen - (unset the status field and let the set_status function will set it based on progress)
</table>

Reply:
Will work on it.